### ISSUE TITLE
Allow user to add more translation fields

### ISSUE THREAD

**Issue by M-Biggles** [jzohrab/lute-v3#198]
So, when a word has multiple meanings, it's hard to cram all the info you have for the different meanings into one translation field, and it's hard to read the multiple meanings when you're hovering over words. The solution with most software has mostly been to be happy with translations for the most common meaning or two without going into depth. It would be very useful to be able to.

Some examples of my current way of doing things:

![1](https://github.com/jzohrab/lute-v3/assets/156795688/6acc9f81-9bc7-471d-b44b-1a2a8b061205)

![2](https://github.com/jzohrab/lute-v3/assets/156795688/99d9fc42-5c72-4b95-841f-43cd626d6586)

![3](https://github.com/jzohrab/lute-v3/assets/156795688/2093ffab-b0a9-4084-8611-46fb581a1283)

I separate the different meanings between the lines of ---------. I have been tagging in the translation field for now until I can tag more than just a single translation field.

My suggestion is to create a system wherein multiple fields are possible. Translation 1, Translation 2, etc. Allow for separate pronunciation and parents for each translation field. If I could press a little + under the Translation field and add the meanings to their own Translation box, it would be very helpful.

Also, it would lead to being able to a mode where the user could select individual (sub)meanings for the contexts they find. (You'd have some issue with editing texts and keeping the selected translation fields, but it should be possible to create a system which updates which word is which.)

**Comment by jzohrab** [comment]
I considered this during the initial design, but decided against it for a few reasons:

* it puts a _very_ large burden on the user.  Of course, users could ignore the "+" button and just dump everything into the main text box.
* some minor technical complexity in the term form
* it would imply some method of "sentence-level notes" which doesn't exist yet (and is a bit tricky -- see #41 )

I look at Lute's dictionary data as a rough blackboard/whiteboard where people scrawl notes about their terms, rather than a thorough dictionary of notes.

I will tag this as wont-fix because it's a fundamental rethink of the data structures and what Lute really _is_ for users :-) but will leave it open -- if someone wanted to fork Lute and add this, hey, it's open source!

**Comment by M-Biggles** [comment]
I'm thinking that I'll see how Lute goes and maybe get another coder to create a sort of 'Linguistics/Advanced mode' with various features like this. We can fork and just add those features. Another idea is a sort of plugin that could be added to Lute.

Having to stuff all the potential definitions into one text field has been a nuisance for a looooong time. It's the same old limitation I find on every single reading platform - this is my one killer feature.

**Comment by jansiebert1** [comment]
Each meaning should also have its own boolean flag whether you know it or not. If you have ten meanings for a word, the main scale of knowledge becomes meaningless. It stops to inform you whether you know a particular meaning or not. The general scale should stay though. 
As far as data structure goes, you could just put all meanings into the same column as json so you do not have to create a new table for meanings.

**Comment by M-Biggles** [comment]
> Each translation field should also have its own boolean flag whether you know it or not. If you have ten translation fields for a word, the main scale of knowledge becomes meaningless. It stops to inform you whether you know a particular meaning or not. The general scale should stay though. As far as data structure goes, you could just put all translation fields into the same column as json so you do not have to create a new table for meanings.

Yeah, it would add a bit of complexity but would also give some pretty fine-grained controls over the whole learning process. Eventually, it would be possible to select single fields to display for particular texts. Word-1 shows translation-2, Word-2 shows translation-1, word-3 shows translation-6B, etc.

One day we'll see it, but we'll have to be patient. If we get a plugin system working, I'd happily pay a computational linguist to get this and a few things implemented.

**Comment by jzohrab** [comment]
Added note to faq: https://luteorg.github.io/lute-faqs/terms/terms-only-have-one-definition.html
